src/main/java/frc/robot/sensors/gyroSensor/GyroSensor.java

```diff
@@ -0,0 +1,83 @@
+package frc.robot.sensors.gyroSensor;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.Logger;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusCode;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.Pigeon2Configuration;
+import com.ctre.phoenix6.hardware.Pigeon2;
+
+// Extends the Subsystem base so we can access the 'periodic' method //
+public class GyroSensor extends SubsystemBase {
+
+  // This is the gyroscope object for the sensor //
+  private Pigeon2 gyro;
+
+  // This is a signal that collects data called 'yawRotations' //
+  // StatusSignals are from Phoenix6 which con constantly retreive certain information //
+  // The 'Double' is a wrapper class for the double paramater //
+  // Read about Status Signals here -> https://v6.docs.ctr-electronics.com/en/stable/docs/api-reference/api-usage/status-signals.html // 
+  private StatusSignal<Double> yawRotations;
+
+  public GyroSensor() {
+
+    // Create the gyroscope sensor with the ID from phoenix tuner x //
+    // If you have a CANiovre attached to the pigeon then you would add a second argument //
+    // The second arguemnt would be the name of the Canivore //
+    gyro = new Pigeon2(GyroConstants.gyroSensor);
+
+    // This is how to apply factory default settings to the gyroscope sensor //
+    // What you are doing here is you are going to get the configurator of the gyroscope to apply settings //
+    // You then use apply to put settings into the configurator //
+    // For now we are going to give a default configuration, or factory settings //
+    gyro.getConfigurator().apply(new Pigeon2Configuration());
+
+    // We set the yaw of the robot to 0 on start up, this will change on more advanced robots //
+    setYaw(0);    
+
+    // 'The StatusSignal<Double>' needs a 'signal' to collect, we gave it the option to collect the signal of the gyrscope yaw //
+    yawRotations = gyro.getYaw();
```

Comment:
This returns the yaw in degrees, not rotations.
https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/hardware/core/CorePigeon2.html#getYaw()

Reply:
okay ill change it